src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -81,12 +110,28 @@ public override void Fail(string message)
         /// </devdoc>
         public override void Fail(string message, string detailMessage)
         {
-            // UIAssert is not enabled.
-            WriteAssert(string.Empty, message, detailMessage);
+            StackTrace stack = new StackTrace(true);
```

Comment:
Adding the stack trace here is just an extra improvement? 

Maybe better like 
`stackTrace = new StackTrace(fNeedFileInfo:true).ToString();`

Reply:
Yes, just an extra improvement